player/playback-engine/src/main/kotlin/com/tidal/sdk/player/playbackengine/ExoPlayerPlaybackEngine.kt

```diff
@@ -802,6 +808,10 @@ internal class ExoPlayerPlaybackEngine(
 
         var errorMessage = "${error.errorCodeName}: ${crawler?.message}"
         while (crawler?.cause != null) {
+            @Suppress("MagicNumber")
+            if ((crawler as? InvalidResponseCodeException)?.responseCode == 416) {
+                playerCache.cache.removeResource(crawler.dataSpec.key!!)
```

Comment:
I think the dataSpec key in question is not in the cache and hence why the player made a request to fetch it. And the request failed with 416 because it used invalid cached range data. I'm not sure how the range data is calculated but I suspect it's from the rest of the cached chunks of the track. So, I suggest we remove all resources related to this media track from the cache. In this PR, we are removing `/mediatracks/Gi...QQ/34.mp4`, but let's remove all `/mediatracks/Gi...QQ/*.mp4`. 

Or alternatively to first validate this fix, we can simply remove all keys from the cache (kinda like a reset) and see if this solves this issue. What do you think?

Reply:
I don't know if our average user would notice that we wipe their entire online playback cache, so we could do that yeah. The other option is probably enough too, but it might be a little more risky since we'd have introduce some regex and make assumptions about how things can look.

Let's go with wiping the entire cache?

Reply:
Yes, let's do that 👍🏻

Reply:
there we go

Reply:
Great - I just pushed a small commit just to make `onPlayerError` not to grow more.